File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/workflow/CommandContext.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.workflow;

import kalix.javasdk.MetadataContext;

/** A value based workflow command context. */
public interface CommandContext extends WorkflowContext, MetadataContext {

  /**
   * The name of the command being executed.
   *
   * @return The name of the command.
   */
  String commandName();

  /**
   * The id of the command being executed.
   *
   * @return The id of the command.
   */
  long commandId();
}

File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/workflow/CommandContext.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.workflow

import kalix.scalasdk.MetadataContext

/** A value based workflow command context. */
trait CommandContext extends WorkflowContext with MetadataContext {

  /**
   * The name of the command being executed.
   *
   * @return
   *   The name of the command.
   */
  def commandName: String

  /**
   * The id of the command being executed.
   *
   * @return
   *   The id of the command.
   */
  def commandId: Long
}
